test/cache.test.js

```diff
@@ -0,0 +1,306 @@
+
+import 'mocha';
+import { expect } from 'chai';
+
+import WS from '../src/echo/ws'
```

Comment:
You need to test the Echo object, not the classes that it uses. So instead of creating Cache, WS, API or WSAPI create Echo instance and pass cache options in second argument of "connect" method